techmatters/hrm
# Added packages/job-errors to resources-import-consumer tsconfig

This PR works on top of the applied changes of https://github.com/techmatters/infrastructure-config/pull/182.
## Description
This PR:
- Upgrades jobs Lambdas to Node 18.
- Removes `node-fetch` dep from `resources-import-consumer`, as it uses global fetch (bundled with Node 18).
- Small fixes around auth, url, to make `resources-import-consumer` successfully call resources internal API.

A message consumed by the `resources-import-consumer` ended in a 200 call to the internal `/resources/import` endpoint.
![Screenshot from 2023-05-01 11-57-27](https://user-images.githubusercontent.com/15805319/235445626-ef5dba1c-1974-43e0-a8e9-5789bf6d8ea0.png)
The normalized resource contained in the message make it all the way and was properly stored in the resources DB.
![Screenshot from 2023-05-01 13-05-29](https://user-images.githubusercontent.com/15805319/235445621-177eea3f-5adb-4679-a215-6f428df315e6.png)


### Verification steps
- Send a message to the [resources-import-consumer-pending sqs queue](https://us-east-1.console.aws.amazon.com/sqs/v2/home?region=us-east-1#/queues/https%3A%2F%2Fsqs.us-east-1.amazonaws.com%2F712893914485%2Fdevelopment-hrm-resources-import-consumer-pending.fifo) with the shape like
  ```
  {"accountSid":"ACd8a2e89748318adf6ddff7df6948deaf","importedResources":[{"id":"RESOURCE_123456","name":"Resource 123456","updatedAt":"2023-04-27T10:03:02.171Z","attributes":{"ResourceStringAttributes":[{"key":"STRING_ATTRIBUTE","value":"VALUE","language":"en-US","info":{"some":"json"}}],"ResourceDateTimeAttributes":[{"key":"DATETIME_ATTRIBUTE","value":"2023-04-27T10:03:02.171Z","info":{"some":"json"}}],"ResourceNumberAttributes":[{"key":"NUMBER_ATTRIBUTE","value":1337,"info":{"some":"json"}}],"ResourceBooleanAttributes":[{"key":"BOOL_ATTRIBUTE","value":true,"info":{"some":"json"}}],"ResourceReferenceStringAttributes":[{"key":"REFERENCE_ATTRIBUTE","value":"REFERENCE_VALUE_2","language":"REFERENCE_LANGUAGE","list":"REFERENCE_LIST_1"}]}}],"batch":{"fromDate":"2023-04-27T10:03:02.171Z","toDate":"2023-04-27T10:03:02.171Z","total":1}}
```
  Note that since id `RESOURCE_123456` is already inserted in the DB, it's better to use a different one.
- Confirm in the logs of the [resources-import-consumer lambda function](https://us-east-1.console.aws.amazon.com/lambda/home?region=us-east-1#/functions/development-hrm-resources-import-consumer?tab=code) that it was called and successfully finished.
- Connect to the resources DB and confirm that a resource with the above given id exists in the `Resources` table.
